src/core_plugins/kibana/public/dashboard/lib/filter_utils.js

```diff
@@ -0,0 +1,51 @@
+import _ from 'lodash';
+
+/**
+ * @typedef {Object} QueryFilter
+ * @property query_string {Object}
+ * @property query_string.query {String}
+ */
+
+export class FilterUtils {
+  /**
+   *
+   * @param filter
+   * @returns {Boolean} True if the filter is of the special query type
+   * (e.g. goes in the query input bar), false otherwise (e.g. is in the filter bar).
+   */
+  static isQueryFilter(filter) {
+    return filter.query && filter.query.query_string && !filter.meta;
+  }
+
+  /**
+   *
+   * @param {SavedDashboard} dashboard
+   * @returns {Array.<Object>} An array of filters stored with the dashboard. Includes
+   * both query filters and filter bar filters.
+   */
+  static getDashboardFilters(dashboard) {
+    return dashboard.searchSource.getOwn('filter');
+  }
+
+  /**
+   * Grabs a saved query to use from the dashboard, or if none exists, creates a default one.
+   * @param {SavedDashboard} dashboard
+   * @returns {QueryFilter}
+   */
+  static getQueryFilterForDashboard(dashboard) {
+    const defaultQueryFilter = { query_string: { query: '*' } };
+    const dashboardFilters = this.getDashboardFilters(dashboard);
+    const dashboardQueryFilter = _.find(dashboardFilters, this.isQueryFilter);
+    return dashboardQueryFilter ? dashboardQueryFilter.query : defaultQueryFilter;
+  }
+
+  /**
+   * Returns the filters for the dashboard that should appear in the filter bar area.
+   * @param {SavedDashboard} dashboard
+   * @return {Array.<Object>} Array of filters that should appear in the filter bar for the
+   * given dashboard
+   */
+  static getFilterBarsForDashboard(dashboard) {
```

Comment:
I think that would then be confusing with `getDashboardFilters`.  I wanted to be explicit that it's extracting "filter bar" filters.  Hence also adding `Filter` suffix to query, I'm just trying to separate them more.  Both are filters, but they are two different types of filters (though hopefully that will change with your project!)

I'm experimenting with having a dashboard_filters class which would keep track of changes made to filters, query and time picker - another reason I wanted to make a distinction between query filter and filter bar filters 

Unfortunately, the introduction of view and edit mode treats all "Filters" (capital F) similarly.  It's even exposed to the user in a dialog (the wording on that may change though). 

<img width="490" alt="screen shot 2017-01-18 at 3 40 45 pm" src="https://cloud.githubusercontent.com/assets/16563603/22082125/7f86473a-dd94-11e6-86ce-e88593085dbb.png">

Agree that once your project has been implemented, hopefully it will remove the need for all this naming confusion!